### PR TITLE
Private events project features code review.

### LINKED ISSUE #1: Morning session
You did a great Job everything works as intended and everything is easy to understand! Keep up the great job

### PR DESCRIPTION
**Pull request by dipbazz**
In this project we created a private events web application where;

- A user can create events.
- A user can attend many events
- An event can be attended by many users.

The models used for this project are:

- `User` model with field `username`
-  `Event` model with field `date`, `description`
-  `Attendence` model for association between `User` as `attendee` and `Event` as `attended_event`

The features of this project in this pull request are:

- On the home page all lists of events will be displayed as past (previous events) and future (upcoming events).
- User can log in to the app.
- User can logout from the app.
- User can view their profile with the list of events they have joined in the past (previous events) and the events they are joined to attend in the future (upcoming events).
- Logged in users can join the event if they haven't joined yet.
- User can create an event.

### CODE REVIEW

File: app/views/events/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+              <h6>Created by:</h6>
+              <p><%= @event.creator.username %></p>
+              <% if @event.attendees.exists? %>
+                <h3>List of Attendees</h3>
+              <ul>
+                <% @event.attendees.each do |attendee| %>
+                  <li><%= attendee.username %></li>
+                <% end %>
+              </ul>
+              <% end %>
```
Comment (oloomoses) [inline app/views/events/show.html.erb]: Remember to your logic into the helpers

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Status: Changes Required
Hello @dipbazz and @Byabasaija, You are doing a very good job on this project :muscle: 
The structure of your pull request is good and very easy to understand :clap: 
Some changes to make before the project can be approved :point_down: 
## Required
Ensure that:
- [x] a logged-in user can attend an event
- [x] a logged-in user is able to logout 
- [x] the user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
- [x] the event's show page displays the list of attendants
##### EventsController
###### `create` action

* [x] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

###### `index` action
* [x] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

##### UsersController
###### `show` action
* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

## Feel free to reach me on slack [ Oloo Moses ] incase of doubts or concerns
# Oloo Moses

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required ♻🛑**

<hr>

Hello there, great job you've done on this project and on the requested changes by the previous Code Reviewers. However, there is still one change you need to make in order to get this pull request approved!

<hr>

- [x] Please fix your Login/Logout logic and functionalities. Try to refactor your sessions logic by creating a specific controller for the sessions (If you see appropriate, you can fix the current logic to make it work properly)

<hr>

Kindly fix the mandatory change above and ask for another re-review 🙏😊

Good luck and happy coding 👨‍💻💻

**Comment by mimipeshy** [comment]
### STATUS :APPROVED :partying_face: :champagne: 

Hey team @Byabasaija  and @dipbazz, you guys have done an amazing job with the feedback given.  This project is therefore approved. Go ahead and merge

All the best in the next projects. Happy Coding :computer: 

Regards
Peris Ndanu

**Pull request merged**